src/coreclr/binder/assemblybindercommon.cpp

```diff
@@ -1396,20 +1338,23 @@ HRESULT AssemblyBinderCommon::BindUsingPEImage(/* in */  AssemblyBinder* pBinder
                 // Attempt was made to load an assembly that has the same name as a previously loaded one. Since same name
                 // does not imply the same assembly, we will need to check the MVID to confirm it is the same assembly as being
                 // requested.
-                //
-                GUID incomingMVID;
-                ZeroMemory(&incomingMVID, sizeof(GUID));
-
-                // If we cannot get MVID, then err on side of caution and fail the
-                // load.
-                IF_FAIL_GO(pIMetaDataAssemblyImport->GetScopeProps(NULL, &incomingMVID));
 
+                GUID incomingMVID;
                 GUID boundMVID;
-                ZeroMemory(&boundMVID, sizeof(GUID));
 
-                // If we cannot get MVID, then err on side of caution and fail the
-                // load.
-                IF_FAIL_GO(bindResult.GetAssembly()->GetMVID(&boundMVID));
+                // GetMVID can throw exception
+                EX_TRY
+                {
+                    pPEImage->GetMVID(&incomingMVID);
+                    bindResult.GetAssembly()->GetPEImage()->GetMVID(&boundMVID);
+                }
+                EX_CATCH
+                {
+                    hr = E_FAIL;
```

Comment:
I have a feeling that this cannot fail. Since we are looking at instances of PEImage we must have already looked into matadata as a part of `AssemblyName::Init`, but I am not completely sure, so I have a catch here.

`GET_EXCEPTION()->GetHR()` sound like a good idea